File: app/Models/WasteCategory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class WasteCategory extends Model
{
    protected $table = 'waste_categories';

    protected $fillable = [

        'main_category',

        'nama_kategori',

        'poin_per_kg'

    ];
}

File: recyclingpoint_mobile/lib/models/waste_category.dart
class WasteCategory {
  final int id;
  final String namaKategori;
  final String mainCategory;
  final int poinPerKg;
  final int hargaPerKg;

  WasteCategory({
    required this.id,
    required this.namaKategori,
    required this.mainCategory,
    required this.poinPerKg,
    required this.hargaPerKg,
  });

  factory WasteCategory.fromJson(Map<String, dynamic> json) {
    return WasteCategory(
      id: json["id"],
      namaKategori: json["nama_kategori"] ?? "",
      mainCategory: json["main_category"] ?? "",
      poinPerKg: json["poin_per_kg"] ?? 0,
      hargaPerKg: json["harga_per_kg"] ?? 0,
    );
  }
}